src/hooks/use-arrow-key-navigation.ts

```diff
@@ -192,7 +191,16 @@ export function useArrowKeyNavigation(
       event.stopPropagation();
     };
 
-    updateTabIndexes(getNavigableElements(), 0, containerVisible && autofocus);
+    const navigableElements = getNavigableElements();
+    // Start focus sequence with previously focused element, if any
+    const initialIndex = lastFocusedItem.current
+      ? navigableElements.indexOf(lastFocusedItem.current)
+      : 0;
+    updateTabIndexes(
+      navigableElements,
+      initialIndex,
+      containerVisible && autofocus,
+    );
```

Comment:
To recap: Every time this effect ran, it reset the selected item to the first one in the list. Now it will reset it to `lastFocusedItem.current`. This is an improvement over the way it worked before, although `lastFocusedItem` doesn't currently get updated if you focus an item a different way, eg using the mouse. In the context of `SelectNext`, this means that if you:

1. Open the list and select an item using the keyboard
2. Re-open the list and select a different item using the mouse
3. Re-open the list using the keyboard

Then the focused item after step (3) is the one that was focused in step (1) rather than step (2).

Reply:
Related to this, the docs for the `autofocus` prop say that focusing is applied when the component is mounted, but the current behavior is that auto-focusing is applied when:

- The component is mounted
- `containerVisible` changes
- Any of the general configuration props for the hook change (we expect this will not happen for the majority of uses)

Reply:
> To recap: Every time this effect ran, it reset the selected item to the first one in the list. Now it will reset it to `lastFocusedItem.current`. This is an improvement over the way it worked before, although `lastFocusedItem` doesn't currently get updated if you focus an item a different way, eg using the mouse. In the context of `SelectNext`, this means that if you:
> 
> 1. Open the list and select an item using the keyboard
> 2. Re-open the list and select a different item using the mouse
> 3. Re-open the list using the keyboard
> 
> Then the focused item after step (3) is the one that was focused in step (1) rather than step (2).

Hmm, good catch.

There's another problem still to be solved, also related with combining mouse and keyboard (see last checkpoint in https://github.com/hypothesis/frontend-shared/issues/1285).

I'll try to address those together.

Also, I have checked that the behavior you describe does not happen with other arrow-key-navigable components, like `Tabs`. If you focus one tab via arrows, then click another tab, then focus away, then focus back, the tab that gets focused is the one you clicked. So at least the bug is limited to the Select.